tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-literals-negative.bal

```diff
@@ -135,3 +135,10 @@ function testXmlNsInterpolationWithinQuery() {
             select xml `<empRecord employeeId="${i}"><inlineNS:id>${i}</inlineNS:id></empRecord>`}
           </empRecords>;`;
 }
+
+function testQueryInXMLTemplateExprNegative() {
+    var _ = xml `<doc>${from string i in ["A", "B"] select i + "C"}</doc>`;
+    xml _ = xml `<doc>${from string i in ["A", "B"] select i + "C"}</doc>`;
+    xml<xml:Element> _ = xml `<doc>${from var i in [1, 2, 3] select i * 2}</doc>`;
+    var _ = xml `<doc>${from int i in 0..<2 select `<foo></foo>`}</doc>`;
```

Comment:
Another one to capture the result type of the whole query. Because L143 negative test only covers the interpolation expected type check.
Maybe an incompatible cast in front of `from`